test/src/mime.js

```diff
@@ -1,8 +1,132 @@
 import { test } from 'socket:test'
 import mime from 'socket:mime'
 
+/**
+ * @param {string[]} left
+ * @param {string[]} right
+ * @return {string[]}
+ */
+function intersection (left, right) {
+  const results = new Set()
+  for (const item of left) {
+    if (right.includes(item)) {
+      // @ts-ignore
+      results.add(item)
+    }
+  }
+
+  for (const item of right) {
+    if (left.includes(item)) {
+      // @ts-ignore
+      results.add(item)
+    }
+  }
+
+  return Array.from(results.entries())
```

Comment:
is it correct that it's an array of entries and not values?

Reply:
`Set.prototype.entries()` returns an [`Iterator`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator) (_Set Entries_) that is an [`Iterable`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols), which we convert to an array with `Array.from()`. This is desired

Reply:
The return value is in JSDoc is wrong than, it's `[number, string][]` on this line and `string[]` in JSDoc

Reply:
maybe we can simplify it to
```js
return left.filter(value => right.includes(value));
```
?